src/types.js

```diff
@@ -59,9 +63,70 @@ export type SpacingAliases = {
   +[spacingAlias: string]: number,
 }
 
+export type DisplayAliases = {
+  +[displayAlias: string]: string,
+}
+
 export type ConfigProviderContext = {|
   +xnReflex: {|
+    +displayAliases: DisplayAliases,
     +spacingAliases: SpacingAliases,
     +base: number,
   |},
 |}
+
+export type OneResolutionGrid = {
+  align?: AlignGrid,
+  children?: React.Node,
+  className?: string,
+  dev?: Dev,
+  justify?: Justify,
+  margin?: Spacing,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginTop?: SpacingValues,
+  padding?: Spacing,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingTop?: SpacingValues,
+  show?: DisplayValues,
+  size?: Size,
+  style?: { [string]: string | number },
+}
+
+export const resolutionProperties = [
+  'align',
+  'justify',
+  'margin',
+  'marginBottom',
+  'marginLeft',
+  'marginRight',
+  'marginTop',
+  'padding',
+  'paddingBottom',
+  'paddingLeft',
+  'paddingRight',
+  'paddingTop',
+  'size',
+]
+
+export type GridProps = {
+  ...OneResolutionGrid,
+  align?: AlignGrid | { [string]: AlignGrid },
+  justify?: Justify | { [string]: Justify },
+  margin?: Spacing | { [string]: Spacing },
+  marginBottom?: SpacingValues | { [string]: SpacingValues },
+  marginLeft?: SpacingValues | { [string]: SpacingValues },
+  marginRight?: SpacingValues | { [string]: SpacingValues },
+  marginTop?: SpacingValues | { [string]: SpacingValues },
+  padding?: Spacing | { [string]: Spacing },
+  paddingBottom?: SpacingValues | { [string]: SpacingValues },
+  paddingLeft?: SpacingValues | { [string]: SpacingValues },
+  paddingRight?: SpacingValues | { [string]: SpacingValues },
+  paddingTop?: SpacingValues | { [string]: SpacingValues },
+  size?: Size | { [string]: Size },
+}
+
+export default undefined
```

Comment:
I dont think you need to make eslint happy with this since you have multiple exports.

Reply:
🙈